### PR TITLE
Georef: crowfly length with speed_factor bug fix

### PR DESCRIPTION
**Pull request by pbougue**
Duration stored in path_items now uses speed_factor, and everything is managed as crowfly path_items
JIRA: http://jira.canaltp.fr/browse/NAVITIAII-1995

### CODE REVIEW

File: source/georef/street_network.cpp [hunk 1]
```diff
@@ -573,8 +575,8 @@ Path create_path(const GeoRef& geo_ref,
         last_transport_carac = transport_carac;
         path_item.way_idx = edge.way_idx;
         path_item.transportation = transport_carac;
-        path_item.duration += edge.duration;
-        p.duration += edge.duration;
+        path_item.duration += edge.duration / speed_factor;
+        p.duration += path_item.duration;
```
Comment (antoine-de) [inline source/georef/street_network.cpp]: oups you're adding to much duration here, not a good thing that the UT pass :wink:

Comment (pbougue) [inline source/georef/street_network.cpp]: I don't understand your comment. The way I see it: we now compute a correct `path_item.duration` first (dealing with speed factor), then add this `path_item.duration` to path duration.
Would have been wrong to add `edge.duration`, yes, but that's not what we do.

Comment (antoine-de) [inline source/georef/street_network.cpp]: nop, you're adding, at each edge, the current total path_item duration to the path. You should also add to the path `edge.duration / speed_factor`, else you'll add some edges multiple times.

Comment (pbougue) [inline source/georef/street_network.cpp]: Oh I see... We have to test it on a path_item with multiple non-null edges. Don't know if it's easy to test.

**Comment by xlqian** [comment]
ok for me

File: source/routing/tests/routing_api_test.cpp [hunk 2]
```diff
@@ -1560,6 +1561,81 @@ BOOST_FIXTURE_TEST_CASE(biking_length_test, streetnetworkmode_fixture<normal_spe
...
+        if (resp.journeys(idx).sections().size() == 1) {
+            journey = resp.journeys(idx);
+        }
+    }
+
+    BOOST_REQUIRE_EQUAL(journey.sections_size(), 1);
+    pbnavitia::Section section = journey.sections(0);
+    BOOST_REQUIRE_EQUAL(section.type(), pbnavitia::SectionType::STREET_NETWORK);
+
+    dump_response(resp, "speed factor length test");
```
Comment (TeXitoi) [inline source/routing/tests/routing_api_test.cpp:1595]: debug dump

File: source/routing/tests/routing_api_test.cpp [hunk 3]
```diff
@@ -1560,6 +1561,81 @@ BOOST_FIXTURE_TEST_CASE(biking_length_test, streetnetworkmode_fixture<normal_spe
...
+
+    std::vector<pbnavitia::PathItem> path_items;
+    for (int s = 0; s < journey.sections_size() ; s++) {
```
Comment (TeXitoi) [inline source/routing/tests/routing_api_test.cpp]: spacing

Comment (TeXitoi) [inline source/routing/tests/routing_api_test.cpp]: and `++s` is better (even if that's the same here).

File: source/routing/tests/routing_api_test.cpp (hunk 2 above)
Comment (pbougue) [inline source/routing/tests/routing_api_test.cpp:1595]: You have to set it explicitly to `true` if you want the dump.
So useless code, but no useless trace. The point is to ease the debug, as on the other tests around.

Comment (TeXitoi) [inline source/routing/tests/routing_api_test.cpp:1595]: OK, then

**Comment by TeXitoi** [comment]
OK for me.

**Pull request merged**